SecurityPkg/Library/Tpm2DeviceLibDTpm/Tpm2PtpSvsmShim.c

```diff
@@ -0,0 +1,123 @@
+/** @file
+  PTP (Platform TPM Profile) CRB (Command Response Buffer) interface shim that switches between
+  SVSM vTPM Ptp and regular Ptp implementations.
+
+  Use TryUseSvsmVTpm () do check for SVSM vTPM presnece and initialzie the shim.
+
+  The SVSM vTPM presence state is cached across library instances.
+
+Copyright (c) 2024 Red Hat
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/Tpm2DeviceLib.h>
+#include <Library/PcdLib.h>
+#include <Library/DebugLib.h>
+#include "Tpm2Ptp.h"
+#include "Tpm2Svsm.h"
+
+/// SVSM vTPM presence state as stored in PcdSvsmVTpmPresence
+/// @{
+#define SVSM_VTPM_PRESENCE_UNKNOWN  0xFF
+#define SVSM_VTPM_PRESNET           0x01
+#define SVSM_VTPM_ABSENT            0x00
+/// @}
+
+static BOOLEAN  mUseSvsmVTpm = FALSE;
+
+/**
+ Test if SVSM vTPM is present and initialize the Ptp-Svsm-Shim.

```

Comment:
What, not how for purpose statements.
```suggestion
 Initializes SVSM vTPM if present, or otherwise uses TCG PTP method.

```